File: frontend/src/api/promoApi.js
import axiosClient from './axiosClient';

const promoApi = {
  /**
   * @param {string} code
   * @param {number} subtotalVnd
   */
  async validate(code, subtotalVnd) {
    const res = await axiosClient.post('/promo-codes/validate', {
      code,
      subtotalVnd,
    });
    return res.data;
  },
};

export default promoApi;

File: mobile/lib/api/promo_api.dart
import 'api_client.dart';

class PromoApi {
  PromoApi(this._client);

  final ApiClient _client;

  Future<Map<String, dynamic>?> validate(String code, int subtotalVnd) async {
    final json = await _client.post('/promo-codes/validate', {
      'code': code,
      'subtotalVnd': subtotalVnd,
    });
    return json['data'] as Map<String, dynamic>?;
  }
}
